File: 258. Add Digits.cpp
//Given an integer num, repeatedly add all its digits until the result has only one digit, and return it.
//Input: num = 38
//Output: 2
//Explanation: The process is
//38 --> 3 + 8 --> 11
//11 --> 1 + 1 --> 2 
//Since 2 has only one digit, return it.
#include <bits/stdc++.h>
using namespace std;

int addDigits(int num)
{
    int rem, temp;
    int sum = 0;

    if (num == 0)
    {
        return 0;
    }
    else
    {
        temp = num;
        while (temp != 0)
        {
            rem = temp % 10;
            sum = sum + rem;
            temp = temp / 10;
        }
        if (sum < 10)    // becz we need single digit
        {
            return sum;
        }
        else
        {
            return addDigits(sum);
        }
    }
}
int main()
{
    int num;
    cout << "Enter number : ";
    cin >> num;

    cout << addDigits(num);
}

File: 258. Add_Digits.js
let addDigits = function (num) {
  if (num === 0 || num < 0) {
    return 0;
  }

  let sum = 0;
  let rem = 0;
  let temp = num;

  while (temp !== 0) {
    rem = temp % 10;
    sum += rem;
    temp = Math.floor(temp / 10);
  }

  if (sum < 10) {
    return sum;
  } else {
    return addDigits(sum);
  }
};

let num = 38;

let result = addDigits(num);

console.log(result);
